apis/r/src/rutilities.h

```diff
@@ -14,6 +14,17 @@ namespace tdbs = tiledbsoma;
 // current build is encoded in TILEDB_VERSION
 #define TILEDB_VERSION TileDB_Version(TILEDB_VERSION_MAJOR,TILEDB_VERSION_MINOR,TILEDB_VERSION_PATCH)
 
+// Create an integer64 object
+inline Rcpp::NumericVector makeInteger64(const std::vector<int64_t>& vec) {
+  size_t n = vec.size();
+
+  Rcpp::NumericVector num(n);
+  std::memcpy(&(num[0]), vec.data(), n*sizeof(double));
+
+  num.attr("class") = "integer64";
+  return(num);
+}
+
 // Convert to a scalar int64_t
 //
 inline int64_t makeScalarInteger64(const double val) {
```

Comment:
as we discussed offline with the three of us, this using double as an intermediate is the correct thing to do, and the straightforward/obvious cast is _not_ the correct thing to do

which is functionally fine, but very non-obvious and will surely confuse anyone else who comes along later

let's get some comments in here why this wrong-looking thing is right-functioning

Reply:
I am sorry but you may be looking at the wrong code getting the wrong impression. Also note that this is not a user facing function, nor is it (AFAICT) called in this PR.

Reply:
Updated: https://github.com/single-cell-data/TileDB-SOMA/pull/1083/files#diff-e450bf4b6b00306a671ed796c0f66adbc26fda96fba948fd8df54d4b2a90243cR19-R26